python/mdvtools/mdvproject.py

```
@@ -1077,6 +1078,166 @@
+                # Print current groups for visibility
+                for group_name in h5.keys():
+                    print(group_name)
+                    
+                # Check for and delete existing group with this name
+                if name in h5:
+                    del h5[name]
+                    print(f"Deleted existing group '{name}' in HDF5 file.")
+                
+                gr = h5.create_group(name)
+            except Exception as e:
+                raise RuntimeError(f"Error managing HDF5 groups for datasource '{name}': {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)